spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/ConsumedMessagesConditions.java

```diff
@@ -41,42 +42,30 @@ static <T> ConsumedMessagesCondition<T> desiredMessageCount(int messageCount) {
 	}
 
 	/**
-	 * Verifies that any of the consumed messages has a payload that equals the specified
-	 * value.
+	 * Verifies that the expected value equals the message payload value of at least one
+	 * consumed message.
 	 * @param expectation the expected value
 	 * @param <T> the type of the message
 	 * @return the condition
 	 */
-	static <T> ConsumedMessagesCondition<T> anyMessageMatchesExpected(T expectation) {
-		return messages -> messages.stream().anyMatch(message -> message.getValue().equals(expectation));
+	static <T> ConsumedMessagesCondition<T> atLeastOneMessageMatches(T expectation) {
+		return messages -> messages.stream().map(Message::getValue).anyMatch(expectation::equals);
 	}
 
 	/**
-	 * Verifies that the consumed messages value contains at all expected values.
+	 * Verifies that each expected value equals the message payload value of at least one
+	 * consumed message.
 	 * @param expectation the expected values
 	 * @param <T> the type of the message
 	 * @return the condition
 	 */
 	@SafeVarargs
 	@SuppressWarnings("varargs")
-	static <T> ConsumedMessagesCondition<T> containsAllExpectedValues(T... expectation) {
+	static <T> ConsumedMessagesCondition<T> atLeastOneMessageMatchesEachOf(T... expectation) {
 		return messages -> {
 			var values = messages.stream().map(Message::getValue).toList();
 			return Stream.of(expectation).allMatch(values::contains);
 		};
 	}
 
-	/**
-	 * Verifies that the consumed messages value contains exactly the expected values.
-	 * @param expectation the expected values
-	 * @param <T> the type of the message
-	 * @return the condition
-	 */
-	@SafeVarargs
-	@SuppressWarnings("varargs")
-	static <T> ConsumedMessagesCondition<T> containsExactlyExpectedValues(T... expectation) {
```

Comment:
I am proposing to drop this variant as I think it may be a bit confusing as well as it will keep looking for matches up to the timeout even though we know that once it passes N messages w/o a match we could short circuit (i.e. it will never match). 

Also, I would like to avoid maintaining an ever growing list of condition permutations. I propose we go out w/ these 3 and then per user request add new conditions when people ask for them. Sound good?